app/src/main/java/com/keylesspalace/tusky/components/occurrence/OccurrenceRepository.kt

```diff
@@ -0,0 +1,108 @@
+package com.keylesspalace.tusky.components.occurrence
+
+import android.util.Log
+import com.keylesspalace.tusky.db.AccountManager
+import com.keylesspalace.tusky.db.AppDatabase
+import kotlinx.coroutines.runBlocking
+import java.util.Calendar
+import javax.inject.Inject
+import kotlin.math.min
+
+class OccurrenceRepository @Inject constructor(private val db: AppDatabase, private val accountManager: AccountManager) {
+    private var lastApiCalls = HashMap<Long, OccurrenceEntity>(13)
+    private var apiCallsCounter = 0
+
+    private val occurrenceDao = db.occurrenceDao()
+
+    fun loadAll(): List<OccurrenceEntity> {
+        val occurrences: List<OccurrenceEntity>
+        runBlocking {
+            occurrences = occurrenceDao.loadAll()
+        }
+
+        return occurrences
+    }
+
+    fun handleApiCallStart(what: String): Long {
+
+        // TODO The account id here could be wrong (for worker tasks for example)
+
+        val occurrence = OccurrenceEntity(
+            accountId = accountManager.activeAccount?.id,
+            type = OccurrenceEntity.Type.APICALL,
+            what = what,
+            startedAt = Calendar.getInstance().time,
+            callTrace = ""
+//            callTrace = OccurrenceEntity.reduceTrace(Throwable().stackTrace),
+        )
+        // TODO all stack traces here have no hint where they might have originated (always ThreadPool)
```

Comment:
This _should_ have a solution.
Otherwise we only have "something happened but you don't know where/why".